File: results/dnd-rest-benchmark/lifecycle-runs/20260730T075409Z_pi_kimi-k2p7-code_ruby-stdlib/lib/player_handbook.rb
# frozen_string_literal: true

# Player's Handbook rules.
module PlayerHandbook
  WIZARD_SPELL_SLOTS = {
    5 => { '1' => 4, '2' => 3, '3' => 2 }
  }.freeze

  def self.spell_slots(payload)
    klass = payload['class']
    level = payload['level']

    return nil unless klass == 'wizard' && level.is_a?(Integer) && level == 5

    slots = WIZARD_SPELL_SLOTS[level]
    return nil unless slots

    {
      'class' => klass,
      'level' => level,
      'slots' => slots
    }
  end

  def self.long_rest(payload)
    level = payload['level']
    hp_current = payload['hp_current']
    hp_max = payload['hp_max']
    hit_dice_spent = payload['hit_dice_spent']
    exhaustion_level = payload['exhaustion_level']

    return nil unless level.is_a?(Integer) && level >= 1 &&
                      hp_current.is_a?(Integer) &&
                      hp_max.is_a?(Integer) && hp_max > 0 &&
                      hit_dice_spent.is_a?(Integer) && hit_dice_spent >= 0 &&
                      exhaustion_level.is_a?(Integer) && exhaustion_level >= 0

    recovered = [level / 2, 1].max
    new_hit_dice_spent = [hit_dice_spent - recovered, 0].max

    {
      'hp_current' => hp_max,
      'hit_dice_spent' => new_hit_dice_spent,
      'exhaustion_level' => [exhaustion_level - 1, 0].max
    }
  end

  def self.equipment_load(payload)
    strength = payload['strength']
    weight = payload['weight']

    return nil unless strength.is_a?(Integer) && strength >= 1 &&
                      weight.is_a?(Integer) && weight >= 0

    capacity = strength * 15

    {
      'capacity' => capacity,
      'weight' => weight,
      'encumbered' => weight > capacity
    }
  end
end

File: results/dnd-rest-benchmark/lifecycle-runs/20260730T150315Z_pi_kimi-k2p7-code_php-stdlib/src/PlayerHandbook.php
<?php

declare(strict_types=1);

/**
 * Player's Handbook helpers: spell slots, rests, and encumbrance.
 */

function getSpellSlots(array $input): array
{
    if (!array_key_exists('class', $input) || !array_key_exists('level', $input)) {
        sendError(400, 'missing fields');
    }

    $class = $input['class'];
    if (!is_string($class) || $class !== 'wizard') {
        sendError(400, 'unsupported class');
    }

    $level = filter_var($input['level'], FILTER_VALIDATE_INT);
    if ($level === false || $level !== 5) {
        sendError(400, 'unsupported level');
    }

    return [
        'class' => 'wizard',
        'level' => 5,
        'slots' => ['1' => 4, '2' => 3, '3' => 2],
    ];
}

function processLongRest(array $input): array
{
    $required = ['level', 'hp_current', 'hp_max', 'hit_dice_spent', 'exhaustion_level'];
    foreach ($required as $key) {
        if (!array_key_exists($key, $input)) {
            sendError(400, 'missing fields');
        }
    }

    $level = filter_var($input['level'], FILTER_VALIDATE_INT);
    $hpCurrent = filter_var($input['hp_current'], FILTER_VALIDATE_INT);
    $hpMax = filter_var($input['hp_max'], FILTER_VALIDATE_INT);
    $hitDiceSpent = filter_var($input['hit_dice_spent'], FILTER_VALIDATE_INT);
    $exhaustionLevel = filter_var($input['exhaustion_level'], FILTER_VALIDATE_INT);

    if ($level === false || $level < 1
        || $hpCurrent === false || $hpCurrent < 0
        || $hpMax === false || $hpMax < 1
        || $hitDiceSpent === false || $hitDiceSpent < 0
        || $exhaustionLevel === false || $exhaustionLevel < 0
    ) {
        sendError(400, 'invalid fields');
    }

    // Long rest restores HP to max and removes one level of exhaustion.
    $newHpCurrent = $hpMax;
    $restoredDice = max(1, intdiv($level, 2));
    $newHitDiceSpent = max(0, $hitDiceSpent - $restoredDice);
    $newExhaustionLevel = max(0, $exhaustionLevel - 1);

    return [
        'hp_current' => $newHpCurrent,
        'hit_dice_spent' => $newHitDiceSpent,
        'exhaustion_level' => $newExhaustionLevel,
    ];
}

function calculateEquipmentLoad(array $input): array
{
    if (!array_key_exists('strength', $input) || !array_key_exists('weight', $input)) {
        sendError(400, 'missing fields');
    }

    $strength = filter_var($input['strength'], FILTER_VALIDATE_INT);
    $weight = filter_var($input['weight'], FILTER_VALIDATE_INT);
    if ($strength === false || $strength < 1 || $weight === false || $weight < 0) {
        sendError(400, 'invalid fields');
    }

    $capacity = $strength * 15;
    $encumbered = $weight > $capacity;

    return [
        'capacity' => $capacity,
        'weight' => $weight,
        'encumbered' => $encumbered,
    ];
}
